docs/development/awk/intro-to-awk/intro-to-awk.md

```diff
@@ -0,0 +1,732 @@
+most important ones are the following:
+
+- The `NF` variable holds the number of fields in the current input record.
+    Notice that each record can have a different number of fields.
+- The `FS` variable is the input field separator. Its default value is the single space
+    but it also matches any sequence of spaces and tabs. Additionally, any number
+	of leading or trailing spaces and tabs is being ignored. If the value of `FS` becomes the null
+	string, then each character in the current line becomes a separate field.
+- The `FILENAME` built-in variable holds the filename of the file currently being
+    processed. Notice that you cannot use `FILENAME` inside the `BEGIN` block because
+	it is not defined yet.
+- The `NR` variable is used for keeping track of the total number of records that have been read so far.
+- The `FNR` variable is used for keeping the total number of records that have been read from the current input file only.
+- The `IGNORECASE` variable is used for telling AWK to ignore case in all of its comparisons or regular expressions.
+- The `ARGC` and `ARGC` variables hold the command line arguments of an AWK program.
+
+### User Defined Variables in AWK
+
+The way you use an AWK variable the first time will define its type. The following
+AWK code uses the `n` variable to count the number of lines that contain a given
+string, which is this case is the word `three`:
+
+    awk '/three/ { n++ }; END { print n+0 }'
+
+The above code says AWK to execute the `n++` code each time there is a match to the
+`/three/` regular expression. When the processing is done, the code in `END` will be
+executed, which just prints the current value of `n` converted to number.
+
+If you provide some input to it, either a filename or user input from the console, you
+will get some output from the AWK code.
+
+## AWK Basics
+
+In this section of the guide you will learn more about the basics of AWK. AWK has support
+for **arrays** and **associative arrays**. It also supports functions and user
+defined functions. AWK programs can have various types of blocks (*patterns*) like
+`BEGIN` and `END` that are executed at the time that is specified by AWK.
+
+### Arrays and Associative Arrays in AWK
+
+This part of the guide will introduce you to AWK **Arrays** and **Associative Arrays**.
+At this point it would be good to know that arrays and associative arrays in AWK are
+exactly the same thing. What logically distinguishes an array from an associative array
+is that the indices of an array are positive integers whereas the keys of an associative
+array are strings. However, AWK uses string keys internally in both cases! Additionally, you
+should use the `delete` statement for removing an element from an array.
+
+Notice that associative arrays are called *hashes* in other programming languages.
+
+#### An example
+
+The following example illustrates associative arrays in AWK using the code found in `arrays.awk`.
+
+{{< file "arrays.awk" awk >}}
```

Comment:
This example relies on using a loop to iterate through the array, but loops aren't described in the next section. Can we switch the order of the sections on arrays and loops so that loops come first?

Reply:
Added a note about the `for` loop.